analysis/topology/sequences.lean

```diff
@@ -0,0 +1,278 @@
+/-
+Copyright (c) 2018 Jan-David Salchow. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jan-David Salchow
+-/
+import analysis.topology.topological_space
+import analysis.metric_space
+import analysis.topology.uniform_space
+import data.set.countable
+import data.real.cau_seq_filter
+
+open filter
+
+/- Sequences in topological spaces.
+ - 
+ - In this file we define sequences in topological spaces and show how they are related to
+ - filters and the topology. In particular, we
+ -
+ - (*) associate a filter with a sequence and show prove equivalence of convergence of the two,
+ - (*) define the sequential closure of a set and prove that it's contained in the closure, 
+ - (*) define a type class "sequential_space" in which closure and sequential closure agree, 
+ - (*) define sequential continuity and show that it coincides with continuity in sequential spaces,
+ - (*) provide an instance that shows that every metric space is a sequential space.
+ -
+ - TODO: There should be an instance that associates a sequential space with a first countable
+ -       space. 
+ - TODO: Sequential compactness should be handeled here.
+ -/
+namespace sequence
+
+universes u v
+variables {X : Type u} {Y : Type v}
+
+def to_filter (x : ℕ → X) : filter X := filter.map x at_top
+
+/- Statements about sequences in general topological spaces. -/
+section topological_space
+variables [topological_space X] [topological_space Y]
+
+/- The notion of convergence of sequences in topological spaces. -/
```

Comment:
Comments like these that precede a definition can be made into documentation comments by using the syntax `/-- ... -/`; then the comment will be shown alongside the type in the editor.